machine/v1/stable.controlplanemachineset.openstack.testsuite.yaml

```diff
@@ -71,6 +71,53 @@ tests:
               platform: OpenStack
               openstack: {}
     expectedError: "spec.template.machines_v1beta1_machine_openshift_io.failureDomains.openstack in body must be of type array: \"object\""
+  - name: Should accept an OpenStack failure domain with an empty failure domain array
+    initial: |
+      apiVersion: machine.openshift.io/v1
+      kind: ControlPlaneMachineSet
+      spec:
+        selector:
+          matchLabels:
+            machine.openshift.io/cluster-api-machine-role: master
+            machine.openshift.io/cluster-api-machine-type: master
+        template:
+          machineType: machines_v1beta1_machine_openshift_io
+          machines_v1beta1_machine_openshift_io:
+            metadata:
+              labels:
+                machine.openshift.io/cluster-api-machine-role: master
+                machine.openshift.io/cluster-api-machine-type: master
+                machine.openshift.io/cluster-api-cluster: cluster
+            spec:
+              providerSpec: {}
+            failureDomains:
+              platform: OpenStack
+              openstack: []
```

Comment:
This seems to be the only way to set the default zones for OpenStack. However, it is incompatible with the `,omitempty` directive for the field.

https://github.com/openshift/api/blob/ea40115b9fa67188f89075ec4e34a9ea346ea123/machine/v1/types_controlplanemachineset.go#L262

How do I encode for "do not substitute values in the providerSpec"?